Fix busy loop in Hydra.Logging

ch1bo opened the pull request:
This was leading to 100% cpu utilization as `flushTBQueue` seems not to block

ch1bo committed:
Fix busy loop in Hydra.Logging

abailly commented on hydra-node/src/Hydra/Logging.hs:
```diff
@@ -77,7 +77,9 @@ withTracer (Verbose name) action = do
+      flushLogs queue >> threadDelay 0.1
 
   flushLogs queue = do
     entries <- atomically $ flushTBQueue queue
```
I misread `flushTBQueue`'s documentation, it says "This function never retries" which obviously means it alwasy returns and never blocks. I am not a big fan of using `threadDelay`, perhaps we should instead use `readTBQueue` in conjunction with `lengthTBQueue` to make sure we block if there's nothing to read, eg. something like
```
do
   len <- lengthTBQueue q
   check $ len > 0
   replicateM (readTBQueue q)
```
instead?

ch1bo replied:
Why not just `forever $ readTBQueue >>= putBSLn . encode`?

ch1bo replied:
Plus the stdout flushing

abailly reviewed (changes requested):
I suggest an alternative to using `threadDelay`.

ch1bo committed:
Use readTBQueue instead of threaDelay

Also use putLBSLn as it is simply writing all the bytes to stdout and
any UTF8 encoded unicode code points from 'encode' will survive this

ch1bo committed:
Remove redundant imports
